test-resources/tests/infrastructure-tests/application.test.ts

```diff
@@ -94,6 +94,34 @@ describe("STS mock infrastructure", () => {
       });
     });
 
+    describe("Domain Names", () => {
+      test("They should always pin ACM certificates to an SSM Parameter version", () => {
+        const ssmWildcardCertRegex =
+          /\{\{resolve\:ssm\:\/\$\{Environment\}\/Platform\/ACM\/AsyncPrimaryZoneWildcardCertificateARN\:[0-9]+\}\}/;
+        const ssmCertRegex =
+          /\{\{resolve\:ssm\:\/\$\{Environment\}\/Platform\/ACM\/[a-zA-Z]+\:[0-9]+\}\}/;
```

Comment:
pedantry: 
https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_PutParameter.html#systemsmanager-PutParameter-request-Name

> Parameter names can include only the following symbols and letters: a-zA-Z0-9_.- 

```suggestion
          /\{\{resolve\:ssm\:\/\$\{Environment\}\/Platform\/ACM\/[a-zA-Z0-9_.-]+\:[0-9]+\}\}/;
```

Although maybe you're expressing here a limitation on how we name the parameters which hold these certificates. 

------ 

sidequest: knowing the allowed characters in an ssm parameter is also useful to create a more generic rule on enforcing versions on ssm parameters.